frontend/src/elm/Main/Models.elm

```diff
@@ -0,0 +1,120 @@
+module Main.Models exposing (..)
+
+
+type alias Model =
+  { majors : List Major
+  , state : Int
+  }
+
+type alias Major =
+  { majorName : String
+  , majorProperty : String
+  , majorSemesters : List Semester
+  , allGradeFull : Int              --총학점
+  , mandatoryGradeFull : Int        --총전공필수학점
+  , electivesGradeFull : Int        --총전공선택학점
+  , generalGradeFull : Int          --총교양학점
+  , allGradeCurr : Int              --현학점
+  , mandatoryGradeCurr : Int        --현전공필수학점
+  , electivesGradeCurr : Int        --현전공선택학점
+  , generalGradeCurr : Int          --현교양학점
+  , remainSubjects : List Subject   --미이수과목
+  , newSemester : Semester
+  }
+
+
+type alias Semester =
+  { year : String
+  , season : String
```

Comment:
I prefer `season` to be `semester`, because
- `Semester` is defined with `year` and `semester` as pk, thus `semester` won't be changed later
- it can be confusing like there's `type Semester` and `type Season`